DataFormats/Headers/include/Headers/NameHeader.h

```diff
@@ -21,45 +21,61 @@
 #include <string>
 #include "Headers/DataHeader.h"
 
-namespace o2 {
-namespace header {
+namespace o2
+{
+namespace header
+{
 
 /// @struct NameHeader
 /// @brief an example data header containing a name of an object as a null terminated char arr.
 /// this is a template! at instantiation the template parameter determines the
 /// size of the held string array.
-/// a caveat with decoding is you have to use header::get<NameHeader<0>>(buffer)
+/// a caveat with decoding is (if length of sting is not known) you have to use header::get<NameHeader<0>>(buffer)
 /// to get it out of a buffer. May improve in the future if enough people complain.
+/// If the compiler decides to pad the end of struct, the length of padding is stored in the last byte
 /// @ingroup aliceo2_dataformats_dataheader
 template <size_t N>
 struct NameHeader : public BaseHeader {
   static const uint32_t sVersion;
   static const o2::header::HeaderType sHeaderType;
   static const o2::header::SerializationMethod sSerializationMethod;
-  NameHeader()
-  : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion)
-  , name()
+  NameHeader() : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion), name()
   {
-    memset(&name[0],'\0',N);
+    //set length of padding in the last byte
+    uint8_t* lastByte = reinterpret_cast<uint8_t*>(this) + sizeof(NameHeader) - 1;
+    *lastByte =
+      reinterpret_cast<const uint8_t*>(this) + sizeof(NameHeader) - reinterpret_cast<const uint8_t*>(&name[N]);
+    //zero string
+    memset(&name[0], '\0', N);
   }
 
-  NameHeader(std::string in)
-  : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion)
-  , name()
+  NameHeader(std::string in) : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion), name()
   {
+    //set length of padding in the last byte
+    uint8_t* lastByte = reinterpret_cast<uint8_t*>(this) + sizeof(NameHeader) - 1;
+    *lastByte =
+      reinterpret_cast<const uint8_t*>(this) + sizeof(NameHeader) - reinterpret_cast<const uint8_t*>(&name[N]);
     // here we actually want a null terminated string
     strncpy(name, in.c_str(), N);
-    name[N-1] = '\0';
+    name[N - 1] = '\0';
   }
 
-private:
+  const char* getName() const { return name; }
+  size_t getNameLength() const
+  {
+    const uint8_t* lastByte = reinterpret_cast<const uint8_t*>(this) + size() - 1;
+    return (lastByte - reinterpret_cast<const uint8_t*>(name)) - *lastByte + 1;
```

Comment:
use `sizeof(NameHeader)`?

Reply:
not sure what you mean here...

Reply:
I though the pointer calculations can be simply replaced by `sizeof(NameHeader) - *lastByte`, which would give the original parameter `N`